helix-core/src/main/java/org/apache/helix/messaging/handling/HelixTaskExecutor.java

```diff
@@ -143,6 +145,7 @@ public HelixTaskExecutor(ParticipantStatusMonitor participantStatusMonitor) {
     _hdlrFtyRegistry = new ConcurrentHashMap<String, MsgHandlerFactoryRegistryItem>();
     _executorMap = new ConcurrentHashMap<String, ExecutorService>();
     _batchMessageExecutorService = Executors.newCachedThreadPool();
```

Comment:
Lets have a default value here of 40 (same as the non batch executor size)  for the thread pool size instead of setting it to cachedThreadPool.